app/src/main/java/com/wireguard/android/activity/TaskerActivity.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright © 2017-2019 WireGuard LLC.
+ * Copyright © 2018-2019 Harsh Shandilya <msfjarvis@gmail.com>. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.wireguard.android.activity
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.view.MenuItem
+import android.view.inputmethod.InputMethodManager
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.getSystemService
+import androidx.databinding.DataBindingUtil
+import com.wireguard.android.BuildConfig
+import com.wireguard.android.R
+import com.wireguard.android.databinding.TaskerActivityBinding
+import com.wireguard.android.model.TunnelManager
+
+class TaskerActivity : AppCompatActivity() {
+
+    val EXTRA_STRING_BLURB = "com.twofortyfouram.locale.intent.extra.BLURB"
+    val EXTRA_BUNDLE = "com.twofortyfouram.locale.intent.extra.BUNDLE"
+
+    private var binding: TaskerActivityBinding? = null
+
+    @Override
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = DataBindingUtil.setContentView(this, R.layout.tasker_activity)
+
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        supportActionBar?.setDisplayShowHomeEnabled(true)
+
+        binding?.taskerSaveButton?.setOnClickListener {
+            saveTaskerAction()
+        }
+
+        val bundle = intent.getBundleExtra(EXTRA_BUNDLE)
+
+        bundle?.let {
+            binding?.taskerTunnelName?.setText(bundle.getString(TunnelManager.TUNNEL_NAME_INTENT_EXTRA))
+            binding?.taskerSecret?.setText(bundle.getString(TunnelManager.INTENT_INTEGRATION_SECRET_EXTRA))
+
+            if (bundle.getString(TunnelManager.TUNNEL_STATE_INTENT_EXTRA) == "${BuildConfig.APPLICATION_ID}.SET_TUNNEL_UP") {
+                binding?.taskerTunnelEnableSwitch?.isChecked = true
+            }
+        }
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            android.R.id.home -> { onBackPressed(); true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    private fun saveTaskerAction() {
+        hideKeyboard()
+        val resultIntent = Intent()
+
+        if (binding?.taskerTunnelName?.text.toString().isBlank()) {
+            setResult(Activity.RESULT_CANCELED, resultIntent)
```

Comment:
```suggestion
            setResult(AppCompatActivity.RESULT_CANCELED, resultIntent)
```

Just how I prefer these